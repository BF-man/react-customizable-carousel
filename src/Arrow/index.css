.rcc-Arrow {
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  z-index: 1;
  cursor: pointer;
  border-radius: 50%;
  background-color: white;

  &--prev {
    top: 50%;
    left: 0;
  }

  &--next {
    top: 50%;
    right: 0;
  }
}
