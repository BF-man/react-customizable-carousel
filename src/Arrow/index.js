import React, { PureComponent } from 'react'

export const ARROW_TYPES = {
  prev: 'prev',
  next: 'next'
}

export class Arrow extends PureComponent {
  get cssClass () {
    const { className, arrowType } = this.props
    const safeArrowType = Object.keys(ARROW_TYPES).includes(arrowType) ? ARROW_TYPES[arrowType] : undefined
    const customClassName = !className ? '' : `${className} ${safeArrowType ? `${className}--${safeArrowType}` : ''}`
    return `rcc-Arrow ${safeArrowType ? `rcc-Arrow--${safeArrowType}` : ''} ${customClassName}`
  }
  render () {
    const { onClick, component, arrowType } = this.props
    const ArrowComponent = component
    return (
      <div className={this.cssClass} >
        <ArrowComponent onClick={onClick} arrowType={arrowType} />
      </div>
    )
  }
}
