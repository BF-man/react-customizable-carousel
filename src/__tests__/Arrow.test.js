import React from 'react'
import renderer from 'react-test-renderer'
import { Arrow } from '../Arrow'

test('Arrow renders component', () => {
  const component = renderer.create(
    <Arrow component={() => <div>Arrow test</div>} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Arrow adds proper arrow type - prev', () => {
  const component = renderer.create(
    <Arrow component={() => <div>Arrow test</div>} arrowType='prev' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Arrow adds proper arrow type - next', () => {
  const component = renderer.create(
    <Arrow component={() => <div>Arrow test</div>} arrowType='next' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Arrow does not add invalid arrow type - test', () => {
  const component = renderer.create(
    <Arrow component={() => <div>Arrow test</div>} arrowType='test' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Arrow adds custom className', () => {
  const component = renderer.create(
    <Arrow component={() => <div>Arrow test</div>} className='customClassName' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Arrow adds proper arrow type - next to customClassName', () => {
  const component = renderer.create(
    <Arrow component={() => <div>Arrow test</div>} arrowType='next' className='customClassName' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Arrow adds proper arrow type - prev to customClassName', () => {
  const component = renderer.create(
    <Arrow component={() => <div>Arrow test</div>} arrowType='prev' className='customClassName' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Arrow does not add invalid arrow type - test', () => {
  const component = renderer.create(
    <Arrow component={() => <div>Arrow test</div>} arrowType='test' className='customClassName' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
