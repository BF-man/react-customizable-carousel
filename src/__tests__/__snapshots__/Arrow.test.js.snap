// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow adds custom className 1`] = `
<div
  className="rcc-Arrow  customClassName "
>
  <div>
    Arrow test
  </div>
</div>
`;

exports[`Arrow adds proper arrow type - next 1`] = `
<div
  className="rcc-Arrow rcc-Arrow--next "
>
  <div>
    Arrow test
  </div>
</div>
`;

exports[`Arrow adds proper arrow type - next to customClassName 1`] = `
<div
  className="rcc-Arrow rcc-Arrow--next customClassName customClassName--next"
>
  <div>
    Arrow test
  </div>
</div>
`;

exports[`Arrow adds proper arrow type - prev 1`] = `
<div
  className="rcc-Arrow rcc-Arrow--prev "
>
  <div>
    Arrow test
  </div>
</div>
`;

exports[`Arrow adds proper arrow type - prev to customClassName 1`] = `
<div
  className="rcc-Arrow rcc-Arrow--prev customClassName customClassName--prev"
>
  <div>
    Arrow test
  </div>
</div>
`;

exports[`Arrow does not add invalid arrow type - test 1`] = `
<div
  className="rcc-Arrow  "
>
  <div>
    Arrow test
  </div>
</div>
`;

exports[`Arrow does not add invalid arrow type - test 2`] = `
<div
  className="rcc-Arrow  customClassName "
>
  <div>
    Arrow test
  </div>
</div>
`;

exports[`Arrow renders component 1`] = `
<div
  className="rcc-Arrow  "
>
  <div>
    Arrow test
  </div>
</div>
`;
